# A Public Health Study

----------------------------------------------------------------------------------------------------------------------

In [1]:
# Import libraries
import numpy as np
import pandas as pd

## Summary

* [Undernourished people's proportion](#chapitre1)

    * [The 10 countries with the highest rates of undernourishment in 2017](#section1_1)
    * [Undernourishment proportion by country in relation to world population in 2017](#section1_2)
    * [Undernourished people in the world in 2017](#section1_3)  
    
* [Food aid](#chapitre2)

    * [The 10 countries receiving the most aid in the world between 2013 and 2017](#section2_1)
    * [The 10 countries receiving the least aid between 2013 and 2017](#section2_2)
    
* [Food availability](#chapitre3)

    * [The 10 countries with the highest food availability in calories per person per day](#section3_1)
    * [The 10 countries with the lowest food availability in calories per person per day](#section3_2)
    
* [World capacity](#chapitre4)

    * [Theoretical number of people in the world who could be fed daily](#section4_1)
    * [Theoretical number of people in the world who could be fed daily with plant-based product ](#section4_2)
    
* [Internal availability and its distribution](#chapitre5)

    * [Repartition rates of internal availability in the world](#section5_1)
    * [Repartition rates of internal availability of cereals](#section5_2)
    
* [Thailand's use of cassava](#chapitre6)

* [Evolution of undernourishment between 2013 and 2017](#chapitre7)

    * [The 10 countries with the highest increase in undernourishment rates](#section7_1)
    * [The 10 countries with the biggest reductions in undernourishment rates](#section7_2)
    
* [Evolution of food availability between 2013 and 2017](#chapitre8)

    * [The 10 countries with the largest increase in food availability (Kcal/person/day)](#section8_1)
    * [The 10 countries with the largest decline in food availability (Kcal/person/day)](#section8_2)
    
* [The 10 countries with the highest loss rates](#chapitre9)

## Undernourished people's proportion <a class="anchor" id="chapitre1"></a>

In [2]:
# Import csv file
sn = pd.read_csv('/Users/antoinebergounhe/Desktop/Openclassroom/Projet 4/DAN-P4-FAO/sous_nutrition.csv')

In [3]:
# Replace interval years with reference year
sn['Année']=sn['Année'].apply(lambda x: x.replace("2012-2014","2013").replace("2013-2015","2014").replace("2014-2016","2015").replace("2015-2017","2016").replace("2016-2018","2017").replace("2017-2019","2018"))

In [4]:
# Replace '<0.1', '' and 'NaN' values with 0 to avoid calculation errors
sn['Valeur']=sn['Valeur'].replace('<0.1','0')
sn['Valeur']=sn['Valeur'].replace('','0')
sn['Valeur']=sn['Valeur'].fillna(0)

In [5]:
# Define the data types for the 'Value' and 'Year' columns as decimal and integer
sn['Valeur']=sn['Valeur'].astype('float')
sn['Année']=sn['Année'].astype('int64')

In [6]:
# Change the value unit to million -> unit to have the same units of measurement between dataframes
sn['Valeur']=sn['Valeur']*1000000

In [7]:
# Import csv file
pop = pd.read_csv('/Users/antoinebergounhe/Desktop/Openclassroom/Projet 4/DAN-P4-FAO/population.csv')

In [8]:
# Modify the value unit. Thousand -> Unit to have the same unit of measurement as the 'sn' dataframe
pop['Valeur']=pop['Valeur']*1000

In [9]:
# Create a dataframe with the population for the year 2017 by country in alphabetical order
pop_2017 = pop[pop['Année'] == 2017].sort_values(by='Valeur').sort_values(['Zone'], ascending=True).reset_index(drop=True)

In [10]:
# Create a dataframe with the population for the year 2013 by country in alphabetical order
pop_2013 = pop[pop['Année'] == 2013].sort_values(by='Valeur').sort_values(['Zone'], ascending=True).reset_index(drop=True)

In [11]:
# Create a dataframe 'snp' consisting of the association of dataframes 'sn' and 'pop' linked by the keys 'zone' and 'year'
snp = pd.merge(sn,pop, on=['Zone', 'Année'])

In [12]:
# Rename the columns in the new dataframe to make them easier to read
snp.rename(columns={'Valeur_x':'Population en sous_nutrition', 'Valeur_y':'Population totale'}, inplace=True)

In [13]:
# Create a new column, each row will be the result of the following mathematical operation
snp['Proportion sous_alimentée']=round((snp['Population en sous_nutrition']/snp['Population totale'])*100,2)

### The 10 countries with the highest rates of undernourishment in 2017 <a class="anchor" id="section1_1"></a>

In [14]:
# Create a dataframe filtered by year '2017', sorted in descending order.
proportion = snp.loc[snp['Année'] == 2017].sort_values('Proportion sous_alimentée', ascending=False).head(10).reset_index(drop=True)
proportion

Zone  Année  \
0                                       Haïti   2017   
1  République populaire démocratique de Corée   2017   
2                                  Madagascar   2017   
3                                     Libéria   2017   
4                                     Lesotho   2017   
5                                       Tchad   2017   
6                                      Rwanda   2017   
7                                  Mozambique   2017   
8                                 Timor-Leste   2017   
9                                 Afghanistan   2017   

   Population en sous_nutrition  Population totale  Proportion sous_alimentée  
0                     5300000.0         10982366.0                      48.26  
1                    12000000.0         25429825.0                      47.19  
2                    10500000.0         25570512.0                      41.06  
3                     1800000.0          4702226.0                      38.28  
4                      800000.0          2091534.0                      38.25  
5                     5700000.0         15016753.0                      37.96  
6                     4200000.0         11980961.0                      35.06  
7                     9400000.0         28649018.0                      32.81  
8                      400000.0          1243258.0                      32.17  
9                    10500000.0         36296113.0                      28.93

**We can see that the country with the highest proportion of undernourished people in 2017 was Haiti, with almost half of its population undernourished (48.26%), closely followed by North Korea with 47.19%, and then Madagascar with 41.06%. Furthermore, of the ten countries with the highest proportion of undernourished people, six are African and three are Asian.**

### Undernourishment proportion by country in relation to world population in 2017  <a class="anchor" id="section1_2"></a>

In [15]:
# Create a global dataframe for the year 2017, which will be used to calculate the proportion of undernourished per country in relation to the global population
prop_mondiale = snp.loc[snp['Année'] == 2017].sort_values('Proportion sous_alimentée', ascending=False)

In [16]:
# Create a world population variable
pop_mondiale = prop_mondiale['Population totale'].sum()

# Create a copy that will be used to calculate the global proportion of undernourished people for each country
proportion_du_pays = prop_mondiale.copy()

# Add a new column with the result of the proportion of undernourished people per country in relation to the world population, rounded to two decimal places
proportion_du_pays['Proportion mondialisée'] = round((proportion_du_pays['Population en sous_nutrition'] / pop_mondiale) * 100, 2)
proportion_du_pays.sort_values('Proportion mondialisée', ascending=False).head(10).reset_index(drop=True)

Zone  Année  \
0                                        Inde   2017   
1                                    Pakistan   2017   
2                                   Indonésie   2017   
3                                     Nigéria   2017   
4                                  Bangladesh   2017   
5                                    Éthiopie   2017   
6                                 Philippines   2017   
7                 République-Unie de Tanzanie   2017   
8                                       Kenya   2017   
9  République populaire démocratique de Corée   2017   

   Population en sous_nutrition  Population totale  Proportion sous_alimentée  \
0                   190100000.0       1.338677e+09                      14.20   
1                    24800000.0       2.079062e+08                      11.93   
2                    23600000.0       2.646510e+08                       8.92   
3                    22800000.0       1.908732e+08                      11.95   
4                    21500000.0       1.596854e+08                      13.46   
5                    21100000.0       1.063999e+08                      19.83   
6                    15700000.0       1.051729e+08                      14.93   
7                    13400000.0       5.466034e+07                      24.52   
8                    11900000.0       5.022114e+07                      23.70   
9                    12000000.0       2.542982e+07                      47.19   

   Proportion mondialisée  
0                    2.52  
1                    0.33  
2                    0.31  
3                    0.30  
4                    0.29  
5                    0.28  
6                    0.21  
7                    0.18  
8                    0.16  
9                    0.16

**Here, if we put the countries on the same denominator, the world population, we find that India has the highest number of undernourished people with 2.52% of the world's population. India is ahead of Pakistan with 0.33% and Indonesia with 0.31%.
Once again, the majority of countries in this ranking come from Asia (6) and Africa (4).**

### Undernourished people in the world in 2017 <a class="anchor" id="section1_3"></a>

In [17]:
# Create a variable to store the number of undernourished people in the world.
nbr_sous_nutrition = prop_mondiale['Population en sous_nutrition'].sum()

# Calculate the rate of undernourishment in the world
taux_sous_nutrition = round(((prop_mondiale['Population en sous_nutrition'].sum())/pop_mondiale*100),2)
print(f"En 2017, le nombre total de personnes en sous-nutrition atteint {nbr_sous_nutrition} individus. Soit {taux_sous_nutrition}% de la population \nmondiale est en situation de sous-nutrition")

En 2017, le nombre total de personnes en sous-nutrition atteint 535700000.0 individus. Soit 7.1% de la population 
mondiale est en situation de sous-nutrition


**As a reminder, India accounts for 2.52% of the 7.1%, or almost one in three undernourished people.**

----------------------------------------------------------------------------------------------------------------------

## Food aid <a class="anchor" id="chapitre2"></a>

In [18]:
# Import csv file
aide = pd.read_csv('/Users/antoinebergounhe/Desktop/Openclassroom/Projet 4/DAN-P4-FAO/aide_alimentaire.csv')

In [19]:
# Filter the dataframe by keeping the 'Cereales' and 'Non-cereales' rows in the 'Product' column.
aide_totale = aide[aide['Produit'].isin(['Céréales','Non-céréales'])]

In [20]:
# Create a new dataframe classified by the countries that received aid between 2013 and 2017.
aide_finale = aide_totale.pivot_table(index='Pays bénéficiaire', columns='Produit', values='Valeur', aggfunc='sum')

# The index is used to classify by country and indirectly to add up the rows for each country.
aide_finale = aide_finale.reset_index()

In [21]:
# Creation of a new column showing the total amount of aid, which facilitates the final calculation
aide_finale['Total'] = aide_finale['Céréales']+aide_finale['Non-céréales']

### The 10 countries receiving the most aid in the world between 2013 and 2017  <a class="anchor" id="section2_1"></a>

In [22]:
# Creation of a column showing the percentage of aid received as a share of total aid.
aide_finale['Pourcentage']= round((aide_finale['Total']/aide_finale['Total'].sum())*100,1)

# Sort by 'Total' in descending order
aide_finale.sort_values('Total', ascending=False).head(10).reset_index(drop=True)

Produit                 Pays bénéficiaire  Céréales  Non-céréales     Total  \
0               République arabe syrienne  625674.0      304244.0  929918.0   
1                                Éthiopie  607691.0       82956.0  690647.0   
2                                   Yémen  491382.0      111860.0  603242.0   
3                           Soudan du Sud  310694.0       36930.0  347624.0   
4                                  Soudan  291939.0       42953.0  334892.0   
5                                   Kenya  231798.0       44620.0  276418.0   
6                              Bangladesh  165269.0        8825.0  174094.0   
7                                 Somalie  119400.0       26939.0  146339.0   
8        République démocratique du Congo  116038.0       28213.0  144251.0   
9                                   Niger  115732.0       22440.0  138172.0   

Produit  Pourcentage  
0               16.9  
1               12.5  
2               10.9  
3                6.3  
4                6.1  
5                5.0  
6                3.2  
7                2.7  
8                2.6  
9                2.5

**The countries with the highest rates of undernourishment are in Asia and Africa. Logically, this ranking includes seven African countries and one Asian country, but it also includes two countries in the Middle East. Among these Middle Eastern countries, Syria is the country that received the most aid in 2017, with 16.9% of global aid, and Yemen is in third place, with 10.9% of global aid.
Surprisingly, there are only three countries with one of the highest numbers of undernourished people: Ethiopia, Kenya and Bangladesh.** 

### The 10 countries receiving the least aid between 2013 and 2017 <a class="anchor" id="section2_2"></a>

In [23]:
# Ranking of countries receiving the least aid between 2013 and 2017
aide_finale.sort_values('Total', ascending=True).head(10).reset_index(drop=True)

Produit     Pays bénéficiaire  Céréales  Non-céréales   Total  Pourcentage
0                     Géorgie       6.0          29.0    35.0          0.0
1                 Timor-Leste      22.0          36.0    58.0          0.0
2                     Vanuatu     331.0          70.0   401.0          0.0
3         Chine, continentale     649.0           0.0   649.0          0.0
4                    Équateur     345.0         336.0   681.0          0.0
5        Sao Tomé-et-Principe     708.0         205.0   913.0          0.0
6                 Tadjikistan     224.0         790.0  1014.0          0.0
7                     Bhoutan     537.0         796.0  1333.0          0.0
8                      Zambie    1349.0         164.0  1513.0          0.0
9                        Iraq     604.0        1195.0  1799.0          0.0

**The presence of certain countries can be explained by the size of their population. Vanuatu, Sao Tome and Principe and Bhutan have populations of less than one million, while Timor-Leste has a population of around 1.2 million. Timor-Leste is also the only country in this ranking to appear among the 10 countries with the highest undernourishment rate (32.17%). The country also received 58 tonnes of food aid between 2013 and 2017.**

----------------------------------------------------------------------------------------------------------------------

## Food availability <a class="anchor" id="chapitre3"></a>

In [24]:
# Import csv file
dispo_alim = pd.read_csv('/Users/antoinebergounhe/Desktop/Openclassroom/Projet 4/DAN-P4-FAO/dispo_alimentaire.csv')

In [25]:
# Replace 'NaN' values with O
dispo_alim = dispo_alim.fillna(0)

In [26]:
# Sort dataframe by country with sum of food availability
dispoKcal = dispo_alim.groupby(['Zone'])['Disponibilité alimentaire (Kcal/personne/jour)'].sum().reset_index()

### The 10 countries with the highest food availability in calories per person per day <a class="anchor" id="section3_1"></a>

In [27]:
# Ranking of countries with highest food availability in calories per person per day
dispoKcal.sort_values('Disponibilité alimentaire (Kcal/personne/jour)', ascending=False).head(10).reset_index(drop=True)

Zone  Disponibilité alimentaire (Kcal/personne/jour)
0               Autriche                                          3770.0
1               Belgique                                          3737.0
2                Turquie                                          3708.0
3  États-Unis d'Amérique                                          3682.0
4                 Israël                                          3610.0
5                Irlande                                          3602.0
6                 Italie                                          3578.0
7             Luxembourg                                          3540.0
8                 Égypte                                          3518.0
9              Allemagne                                          3503.0

**The three countries with the highest food availability in calories are Austria (3770 calories per person per day), Belgium (3737 calories) and Turkey (3708 calories). Of the ten countries with the highest food availability, six are in Europe, three in the Middle East/Africa and the United States of America.**

### The 10 countries with the lowest food availability in calories per person per day <a class="anchor" id="section3_2"></a>

In [28]:
# Rank countries with lowest food availability in calories per person per day
dispoKcal.sort_values('Disponibilité alimentaire (Kcal/personne/jour)', ascending=True).head(10).reset_index(drop=True)

Zone  \
0                   République centrafricaine   
1                                      Zambie   
2                                  Madagascar   
3                                 Afghanistan   
4                                       Haïti   
5  République populaire démocratique de Corée   
6                                       Tchad   
7                                    Zimbabwe   
8                                     Ouganda   
9                                 Timor-Leste   

   Disponibilité alimentaire (Kcal/personne/jour)  
0                                          1879.0  
1                                          1924.0  
2                                          2056.0  
3                                          2087.0  
4                                          2089.0  
5                                          2093.0  
6                                          2109.0  
7                                          2113.0  
8                                          2126.0  
9                                          2129.0

**We see that the country with the lowest food availability is the Central African Republic, with 1879 calories per person per day, half that of Austria. Logically, the countries with the lowest food availability include those with high levels of malnutrition, such as Madagascar, Timor-Leste, Afghanistan, North Korea, Haiti and Chad.
Most of the countries in this ranking are on the African continent.**

----------------------------------------------------------------------------------------------------------------------

## World capacity <a class="anchor" id="chapitre4"></a>

### Theoretical number of people in the world who could be fed daily<a class="anchor" id="section4_1"></a>

In [29]:
# Creation of a new dataframe combining food availability and population for 2013
dispo_mondiale = pd.merge(dispoKcal,pop_2013, on='Zone', how='right')

In [30]:
# Replaced 'NaN' values with O
dispo_mondiale = dispo_mondiale.fillna(0)

In [31]:
# Created a new column with food availability in calories per day per country
dispo_mondiale['dispo Kcal/jour'] = dispo_mondiale['Disponibilité alimentaire (Kcal/personne/jour)']*dispo_mondiale['Valeur']

In [32]:
# Calculated the theoretical number of people in the world that could be fed per day, stored in a variable
disponibilité_mondiale_jour = round((dispo_mondiale['dispo Kcal/jour'].sum()/2500),0)

# Calculate the percentage of the theoretical number of people in the world that can be fed per day, stored in another variable
taux_mondiale_jour = round(((dispo_mondiale['dispo Kcal/jour'].sum()/2500)/(dispo_mondiale['Valeur'].sum()))*100,2)
print(f"En 2013, sur une base de 2500 calories par jour le nombre théorique de personnes qui pourraient être nourries \ns'élèvent à {disponibilité_mondiale_jour} personnes par jour, soit une capacité à nourrir {taux_mondiale_jour}% de la population mondiale par jour")

En 2013, sur une base de 2500 calories par jour le nombre théorique de personnes qui pourraient être nourries 
s'élèvent à 8017745251.0 personnes par jour, soit une capacité à nourrir 111.19% de la population mondiale par jour


**Now, if we keep the same base of calories per day, 2500. We can see that we could feed more people than the world's population. However, one in four countries has a calorie deficit and the undernourishment rate is 7.1%.**

### Theoretical number of people in the world who could be fed daily with plant-based product <a class="anchor" id="section4_2"></a>

In [33]:
# Create a dataframe sorted by plant-based products and their food availability by country
dispoVegetal = dispo_alim.loc[dispo_alim['Origine'] == 'vegetale'].groupby('Zone')['Disponibilité alimentaire (Kcal/personne/jour)'].sum().reset_index()

In [34]:
# Create a dataframe combining the dataframe created above and the population per country in 2013
dispo_vege_mondial = pd.merge(dispoVegetal,pop_2013, on='Zone', how='right')

In [35]:
# Replace the value 'NaN' with 0
dispo_vege_mondial = dispo_vege_mondial.fillna(0)

In [36]:
# Create a new column with the food availability in calories per day per country
dispo_vege_mondial['dispo Kcal/jour'] = dispo_vege_mondial['Disponibilité alimentaire (Kcal/personne/jour)']*dispo_vege_mondial['Valeur']

In [37]:
# Calculate the theoretical number of people in the world that could be fed plant-based products daily, stored in a variable
disponibilité_vege_mondiale_jour = round((dispo_vege_mondial['dispo Kcal/jour'].sum()/2500),0)

# Calculate the percentage of the theoretical number of people in the world that could be fed daily with plant-based products, stored in another variable
taux_vege_mondiale_jour = round(((dispo_vege_mondial['dispo Kcal/jour'].sum()/2500)/(dispo_vege_mondial['Valeur'].sum()))*100,2)
print(f"In 2013, based on 2500 calories per day, the theoretical number of people who could be fed exclusively on plant-based foods was {disponibilité_vege_mondiale_jour} people per day, or {taux_vege_mondiale_jour}% of the world's population.")

In 2013, based on 2500 calories per day, the theoretical number of people who could be fed exclusively on plant-based foods was 6602669561.0 people per day, or 91.57% of the world's population.


----------------------------------------------------------------------------------------------------------------------

## Internal availability and its distribution <a class="anchor" id="chapitre5"></a>

In [38]:
# Create a dataframe sorted by country with the columns we are interested in
disporepart = dispo_alim.groupby(['Zone'])[['Disponibilité intérieure', 'Aliments pour animaux', 'Autres Utilisations', 'Nourriture', 'Pertes', 'Semences', 'Traitement']].sum().reset_index()

In [39]:
# Create a list of the columns that will be used to calculate the percentages
liste = ['Aliments pour animaux', 'Autres Utilisations', 'Nourriture', 'Pertes', 'Semences', 'Traitement']

# Create a loop that takes the columns in the list and calculates the percentages of each column in relation to internal availability
for col in liste:
    
    # Use the fstring to name these new columns
    disporepart[f"Part {col}"] = round((disporepart[col]/disporepart['Disponibilité intérieure'])*100,0)

### Repartition rates of internal availability in the world <a class="anchor" id="section5_1"></a>

In [40]:
# Create a 'for' loop to calculate the rate for the three columns involved
# The string is created using the 'part_...' variables
for col in ['Aliments pour animaux', 'Pertes', 'Nourriture']:
    part = round((disporepart[col].sum() / disporepart['Disponibilité intérieure'].sum()) * 100, 0)
    if col == 'Aliments pour animaux':
        part_animaux = part
    elif col == 'Pertes':
        part_pertes = part
    elif col == 'Nourriture':
        part_humains = part
       
print(f"In 2013, at the global level, the proportion of food allocated to animal feed was {part_animaux}%, the waste rate was {part_pertes}% and the proportion allocated to human food was {part_humains}%.")

In 2013, at the global level, the proportion of food allocated to animal feed was 13.0%, the waste rate was 5.0% and the proportion allocated to human food was 50.0%.


### Repartition rates of internal availability of cereals <a class="anchor" id="section5_2"></a>

In [41]:
# Create a list of all the grains to use as a 'filter'
liste_céréales=['Avoine', 'Blé', 'Céréales, Autres', 'Maïs', 'Millet', 'Orge', 'Riz (Eq Blanchi)', 'Seigle', 'Sorgho']

In [42]:
# Create a dataframe that keeps the values in the cereal list in the 'Product' column
dispo_céréales=dispo_alim[dispo_alim['Produit'].isin(liste_céréales)]
dispo_céréales=dispo_céréales.groupby(by='Zone')[['Aliments pour animaux', 'Autres Utilisations', 'Disponibilité intérieure','Nourriture', 'Pertes', 'Semences', 'Traitement']].sum().reset_index()

In [43]:
# Create a list to be used in a 'for' loop
liste2 = ['Aliments pour animaux', 'Autres Utilisations', 'Nourriture', 'Pertes', 'Semences', 'Traitement']

# Create a loop that calculates the percentage for all columns in the list in relation to internal availability
for column in liste2:
    dispo_céréales[f"Part {column}"] = round((dispo_céréales[column]/dispo_céréales['Disponibilité intérieure'])*100,0)

In [44]:
# Create a 'for' loop to calculate the rate for the three columns concerned
for col in ['Aliments pour animaux', 'Pertes', 'Nourriture']:
    
    # The loop will calculate each rate at each iteration and store the result in the appropriate variable
    part = round((dispo_céréales[col].sum() / dispo_céréales['Disponibilité intérieure'].sum()) * 100, 0)
    
    # The string is created using the 'part_...' variables.
    if col == 'Aliments pour animaux':
        part_animaux = part
    elif col == 'Pertes':
        part_pertes = part
    elif col == 'Nourriture':
        part_humains = part
       
print(f"In 2013, global cereal use was distributed as follows: the share of cereals used for animal feed was {part_animaux}%, the share of cereals lost was {part_pertes}%, and the share used for human food was {part_humains}%.")

In 2013, global cereal use was distributed as follows: the share of cereals used for animal feed was 36.0%, the share of cereals lost was 4.0%, and the share used for human food was 43.0%.


----------------------------------------------------------------------------------------------------------------------

## Thailand's use of cassava <a class="anchor" id="chapitre6"></a>

In [45]:
# Use the 'dispo_alim' dataframe already created, filtered to select only cassava in Thailand
manioc=dispo_alim.loc[dispo_alim['Produit']=='Manioc'].loc[dispo_alim['Zone']=='Thaïlande']

In [46]:
# Create a dataframe that merges the 'manioc' and 'snp' dataframes
# This merge will provide all data related to the use of cassava in Thailand
thai_manioc = pd.merge(manioc,snp, on=['Zone'])
thai_manioc= thai_manioc.loc[thai_manioc['Année']==2013]

In [47]:
# Create a dataframe with all the data for Thailand that are in the 'dispo_alim' dataframe
# This dataframe will be useful for calculating the share of cassava production in the country's total production
prod_thailande=dispo_alim.loc[dispo_alim['Zone']=='Thaïlande']

In [48]:
# Calculation of the percentage of cassava consumption
part_manioc = round((thai_manioc['Production']/(prod_thailande['Production'].sum()))*100,2)

part_export_manioc = round((thai_manioc['Exportations - Quantité']/(prod_thailande['Exportations - Quantité'].sum()))*100,2)

taux_prod_export = round((thai_manioc['Exportations - Quantité']/thai_manioc['Production'])*100,2)

print(f"In 2013, cassava accounted for {float(part_manioc)}% of Thailand's food production. n\Moreover, with {float(taux_prod_export)}% of production exported, cassava makes up {float(part_export_manioc)}% of the country's exports.")

In 2013, cassava accounted for 14.98% of Thailand's food production. n\Moreover, with 83.41% of production exported, cassava makes up 50.0% of the country's exports.


----------------------------------------------------------------------------------------------------------------------

## Evolution of undernourishment between 2013 and 2017 <a class="anchor" id="chapitre7"></a>

In [49]:
# Create a dataframe with the 'snp' dataframe (under nutrition and population) filtered by the years '2013' and '2017'
diff_sn= snp.loc[snp['Année'].isin([2013, 2017])]

In [50]:
# Create a double-entry table from the created dataframe.
# The table will be sorted by country, the columns will be '2013' and '2017', the values will be the undernourishment rate per year per country
diff_sous_nut = diff_sn.pivot(index='Zone', columns='Année', values='Proportion sous_alimentée')

### The 10 countries with the highest increase in undernourishment rates <a class="anchor" id="section7_1"></a>

In [51]:
# Create a new column whose values reflect the difference between the percentage of undernourishment between '2013' and '2017'
diff_sous_nut['Évolution'] = (diff_sous_nut[2017] - diff_sous_nut[2013])

# Ranking of the 10 countries with an increase in their undernourishment rate
diff_sous_nut.sort_values(['Évolution'], ascending=False).head(10)

Année                                        2013   2017  Évolution
Zone                                                               
Lesotho                                     14.79  38.25      23.46
Venezuela (République bolivarienne du)       4.70  27.21      22.51
Mozambique                                  23.08  32.81       9.73
Rwanda                                      25.90  35.06       9.16
Eswatini                                     9.20  17.78       8.58
Madagascar                                  33.10  41.06       7.96
République populaire démocratique de Corée  41.32  47.19       5.87
Mauritanie                                   7.86  11.68       3.82
Nigéria                                      8.56  11.95       3.39
Gambie                                      10.18  13.55       3.37

**Between 2013 and 2017, two countries saw a significant increase in their undernourishment rates: Lesotho, with an increase of 23.46 percentage points from 14.79% in 2013 to 38.25% in 2017. And Venezuela, with an increase of 22.51 percentage points from 4.70% to 27.21%.
We can noted that none of these countries are among the 10 countries that receive the most or the least food aid.**

### The 10 countries with the biggest reductions in undernourishment rates <a class="anchor" id="chapitre7_2"></a>

In [52]:
# Ranking of the 10 countries where the undernourishment rate has decreased
# ascending=True' is used to rank the column in ascending order
diff_sous_nut.sort_values(['Évolution'], ascending=True).head(10)

Année                             2013   2017  Évolution
Zone                                                    
Namibie                          22.39  12.49      -9.90
Angola                           29.21  19.45      -9.76
Éthiopie                         27.47  19.83      -7.64
Liban                            11.84   5.87      -5.97
Équateur                         14.64   8.94      -5.70
Honduras                         19.32  13.79      -5.53
Sierra Leone                     32.05  26.71      -5.34
Pakistan                         16.26  11.93      -4.33
Ouzbékistan                       6.68   2.82      -3.86
Bolivie (État plurinational de)  18.97  15.19      -3.78

**This table shows the top ten positive changes in undernourishment in specific countries, including Namibia with a decrease of -9.90 percentage points from 22.39% to 12.49%, Angola with a decrease of 9.76 points from 29.21% to 19.45%, and Ethiopia with a decrease of 7.64 points. It is interesting to note that Ethiopia is the second country that received the most food aid between 2013 and 2017, with 12.5% of international aid. In contrast, Ecuador, which saw its undernourishment rate fall by 5.70 points, ranks 5th among the countries that received the least food aid between 2013 and 2017.** 

----------------------------------------------------------------------------------------------------------------------

## Evolution of food availability between 2013 and 2017 <a class="anchor" id="chapitre8"></a>

In [53]:
# Import csv file
dispo_alim_2017 = pd.read_csv('/Users/antoinebergounhe/Desktop/Openclassroom/Projet 4/DAN-P4-FAO/dispo_alimentaire_2017.csv')

In [54]:
# Create a dataframe for 2017 in the form of a double-entry table using the 'pivot_table' method
# The rows are the countries, the columns are 'food availability 2017' and 'domestic availability 2017'
diff_dispo_2017 = dispo_alim_2017.pivot_table(index='Zone', columns='Élément', values='Valeur', aggfunc='sum')

# Create another dataframe for the year 2013 using the '.groupby' method. The dataframe will be sorted by zone using the sum of the columns selected with the '.sum()' function
diff_dispo_2013 = dispo_alim.groupby(['Zone'])['Disponibilité alimentaire (Kcal/personne/jour)'].sum()

In [55]:
# Create a dataframe to link the 2013 data to the 2017 data
diff_dispo = pd.merge(diff_dispo_2017,diff_dispo_2013, on=['Zone'])

# Columns renamed to avoid duplication between 2013 and 2017
diff_dispo = diff_dispo.rename(columns={'Disponibilité alimentaire (Kcal/personne/jour)_x':'Disponibilité alimentaire 2017 (Kcal/personne/jour)','Disponibilité alimentaire (Kcal/personne/jour)_y':'Disponibilité alimentaire 2013 (Kcal/personne/jour)'})

# Removal of the indoor availability column, as it is useless in this case
diff_dispo = diff_dispo.drop(['Disponibilité intérieure'],axis=1)

# Addition of the 'Food evolution' column to reflect the increase or decrease in food availability between 2013 and 2017
diff_dispo['Évolution alimentaire'] = round(diff_dispo['Disponibilité alimentaire 2017 (Kcal/personne/jour)'] - diff_dispo['Disponibilité alimentaire 2013 (Kcal/personne/jour)'],0)

### The 10 countries with the largest increase in food availability (Kcal/person/day) <a class="anchor" id="section8_1"></a>

In [56]:
diff_dispo.sort_values(['Évolution alimentaire'], ascending=False).head(10)

Disponibilité alimentaire 2017 (Kcal/personne/jour)  \
Zone                                                                                         
Ouzbékistan                                                                      3197.02     
Tadjikistan                                                                      2618.33     
Chypre                                                                           3038.48     
Chine - RAS de Macao                                                             3298.72     
Guyana                                                                           3130.54     
Éthiopie                                                                         2484.90     
République démocratique populaire lao                                            2768.48     
Zambie                                                                           2232.15     
Namibie                                                                          2464.90     
Islande                                                                          3651.71     

                                       Disponibilité alimentaire 2013 (Kcal/personne/jour)  \
Zone                                                                                         
Ouzbékistan                                                                       2759.0     
Tadjikistan                                                                       2199.0     
Chypre                                                                            2646.0     
Chine - RAS de Macao                                                              2915.0     
Guyana                                                                            2762.0     
Éthiopie                                                                          2129.0     
République démocratique populaire lao                                             2455.0     
Zambie                                                                            1924.0     
Namibie                                                                           2166.0     
Islande                                                                           3381.0     

                                       Évolution alimentaire  
Zone                                                          
Ouzbékistan                                            438.0  
Tadjikistan                                            419.0  
Chypre                                                 392.0  
Chine - RAS de Macao                                   384.0  
Guyana                                                 369.0  
Éthiopie                                               356.0  
République démocratique populaire lao                  313.0  
Zambie                                                 308.0  
Namibie                                                299.0  
Islande                                                271.0

**We can see that two Central Asian countries have seen the biggest increases in 4 years. Uzbekistan and Tajikistan have increased their food availability by more than 400 calories per person per day, with increases of 438 and 419 respectively, meaning that Tajikistan has moved from a food deficit to a food surplus, as has Laos, which has the seventh best increase at 313 calories. Ethiopia and Namibia are also approaching 2,500 calories per person per day, with increases of 356 and 299 respectively. Zambia, which was the country with the second lowest food availability in 2013, saw its figure increase by 308 calories per person per day, from 1924 calories to 2232 calories, despite being the ninth country to receive the least food aid over the same period.**

### The 10 countries with the largest decline in food availability (Kcal/person/day) <a class="anchor" id="section8_2"></a>

In [57]:
diff_dispo.sort_values(['Évolution alimentaire'], ascending=True).head(10)

Disponibilité alimentaire 2017 (Kcal/personne/jour)  \
Zone                                                                                          
Maldives                                                                          2268.53     
Jordanie                                                                          2739.50     
Mozambique                                                                        1931.01     
Lesotho                                                                           2181.03     
Venezuela (République bolivarienne du)                                            2354.63     
Yémen                                                                             1990.90     
Oman                                                                              2935.84     
Émirats arabes unis                                                               3074.52     
Gabon                                                                             2640.50     
Égypte                                                                            3343.69     

                                        Disponibilité alimentaire 2013 (Kcal/personne/jour)  \
Zone                                                                                          
Maldives                                                                           2733.0     
Jordanie                                                                           3099.0     
Mozambique                                                                         2282.0     
Lesotho                                                                            2528.0     
Venezuela (République bolivarienne du)                                             2633.0     
Yémen                                                                              2217.0     
Oman                                                                               3140.0     
Émirats arabes unis                                                                3275.0     
Gabon                                                                              2826.0     
Égypte                                                                             3518.0     

                                        Évolution alimentaire  
Zone                                                           
Maldives                                               -464.0  
Jordanie                                               -360.0  
Mozambique                                             -351.0  
Lesotho                                                -347.0  
Venezuela (République bolivarienne du)                 -278.0  
Yémen                                                  -226.0  
Oman                                                   -204.0  
Émirats arabes unis                                    -200.0  
Gabon                                                  -186.0  
Égypte                                                 -174.0

**The Maldives is the country with the biggest drop in food availability, -464 calories per day per person, taking the country from a calorie surplus to a deficit, falling below the 2,500 calorie mark. Other countries have suffered the same fate, with Lesotho falling from 2,528 to 2,181 and Venezuela from 2,633 in 2013 to 2,354 per person per day in 2017. It is interesting to note that Mozambique, Lesotho and Venezuela are also the top three countries that have seen the biggest increase in their undernourishment rates; on the other hand, despite these many countries seeing a decrease, only Yemen received significant food aid, with 10.9% of international aid.**

In [58]:
# The number of countries in 2013 with a value above 2,500 calories per person per day is stored in a variable
plus_2500_2013 = dispoKcal.sort_values(by=['Disponibilité alimentaire (Kcal/personne/jour)'], ascending=False).query("`Disponibilité alimentaire (Kcal/personne/jour)` > 2500").shape[0]
moins_2500_2013 = dispoKcal.sort_values(by=['Disponibilité alimentaire (Kcal/personne/jour)'], ascending=True).query("`Disponibilité alimentaire (Kcal/personne/jour)` < 2500").shape[0]

# The number of countries in 2017 with a value above 2,500 calories per person per day is stored in a variable
plus_2500_2017 = diff_dispo_2017.sort_values(by=['Disponibilité alimentaire (Kcal/personne/jour)'], ascending=False).query("`Disponibilité alimentaire (Kcal/personne/jour)` > 2500").shape[0]
moins_2500_2017 = diff_dispo_2017.sort_values(by=['Disponibilité alimentaire (Kcal/personne/jour)'], ascending=True).query("`Disponibilité alimentaire (Kcal/personne/jour)` < 2500").shape[0]

print(f"In 2013, the number of countries with food availability in calories per person per day above 2500 was {plus_2500_2013}, and the number with availability below 2500 was {moins_2500_2013}. In 2017, the number of countries with a value above 2500 is {plus_2500_2017} and {moins_2500_2017} countries have a lower value.")

In 2013, the number of countries with food availability in calories per person per day above 2500 was 132, and the number with availability below 2500 was 42. In 2017, the number of countries with a value above 2500 is 143 and 37 countries have a lower value.


**In 2013, we found that 24% of countries did not have enough food availability (in calories) to be considered healthy, with 132 compared to 42. In 2017, the number of countries with sufficient food availability has increased and the number with insufficient food availability has decreased, so in four years we have gone from 24% to 20% of countries with insufficient food availability.**

----------------------------------------------------------------------------------------------------------------------

## The 10 countries with the highest loss rates<a class="anchor" id="chapitre9"></a>

In [59]:
# Create a dataframe with countries ranked from highest to lowest using the '.sort_values' method
class_perte = disporepart.sort_values(['Part Pertes'], ascending=False).head(10).reset_index(drop=True)
class_perte

Zone  Disponibilité intérieure  Aliments pour animaux  \
0  Sierra Leone                    6570.0                  901.0   
1         Ghana                   38998.0                 4878.0   
2        Angola                   32029.0                 9259.0   
3      Cameroun                   26222.0                 2166.0   
4     Dominique                     141.0                   20.0   
5        Guinée                    8404.0                  648.0   
6          Togo                    4095.0                  151.0   
7      Bulgarie                    9458.0                 1055.0   
8        Malawi                   19877.0                 4009.0   
9       Uruguay                    4988.0                 1239.0   

   Autres Utilisations  Nourriture  Pertes  Semences  Traitement  \
0                225.0      3070.0  2191.0      55.0       126.0   
1               3768.0     22482.0  7442.0      77.0       429.0   
2                451.0     15059.0  4799.0     337.0      2121.0   
3               1027.0     15259.0  3981.0     337.0      3451.0   
4                 24.0        68.0    20.0       4.0         5.0   
5                371.0      5711.0  1074.0     116.0       484.0   
6                 54.0      3186.0   476.0     152.0       120.0   
7                535.0      5111.0  1090.0     357.0      1312.0   
8               1296.0      8607.0  2428.0     354.0      3191.0   
9                125.0      2658.0   531.0     183.0       280.0   

   Part Aliments pour animaux  Part Autres Utilisations  Part Nourriture  \
0                        14.0                       3.0             47.0   
1                        13.0                      10.0             58.0   
2                        29.0                       1.0             47.0   
3                         8.0                       4.0             58.0   
4                        14.0                      17.0             48.0   
5                         8.0                       4.0             68.0   
6                         4.0                       1.0             78.0   
7                        11.0                       6.0             54.0   
8                        20.0                       7.0             43.0   
9                        25.0                       3.0             53.0   

   Part Pertes  Part Semences  Part Traitement  
0         33.0            1.0              2.0  
1         19.0            0.0              1.0  
2         15.0            1.0              7.0  
3         15.0            1.0             13.0  
4         14.0            3.0              4.0  
5         13.0            1.0              6.0  
6         12.0            4.0              3.0  
7         12.0            4.0             14.0  
8         12.0            2.0             16.0  
9         11.0            4.0              6.0

**In 2013, a country like Sierra Leone lost 33% of its internal availability. Ghana, in second place, lost 19% of its availability and Angola 15%. Of the ten countries with the highest loss rates, seven are African.**